ReactiveCocoa/Association.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  Association.swift
+//  Rex
+//
+//  Created by Neil Pankey on 6/19/15.
+//  Copyright (c) 2015 Neil Pankey. All rights reserved.
+//
+
+import ReactiveSwift
+
+/// Attaches a `MutableProperty` value to the `host` object using KVC to get the initial
+/// value and write subsequent updates from the property's producer. Note that `keyPath`
+/// is a `StaticString` because it's pointer value is used as key value when associating
+/// the property.
+///
+/// This can be used as an alternative to `DynamicProperty` for creating strongly typed
+/// bindings on Cocoa objects.
+public func associatedProperty(_ host: AnyObject, keyPath: StaticString) -> MutableProperty<String> {
+	let initial: (AnyObject) -> String  = { host in
+		host.value(forKeyPath: String(describing: keyPath)) as? String ?? ""
+	}
+	let setter: (AnyObject, String) -> () = { host, newValue in
+		host.setValue(newValue, forKeyPath: String(describing: keyPath))
+	}
+	return associatedProperty(host, key: keyPath.utf8Start, initial: initial, setter: setter)
+}
+
+/// Attaches a `MutableProperty` value to the `host` object using KVC to get the initial
+/// value and write subsequent updates from the property's producer. Note that `keyPath`
+/// is a `StaticString` because it's pointer value is used as key value when associating
+/// the property.
+///
+/// This can be used as an alternative to `DynamicProperty` for creating strongly typed
+/// bindings on Cocoa objects.
+public func associatedProperty<T: AnyObject>(_ host: AnyObject, keyPath: StaticString, placeholder: () -> T) -> MutableProperty<T> {
+	let setter: (AnyObject, T) -> () = { host, newValue in
+		host.setValue(newValue, forKeyPath: String(describing: keyPath))
+	}
+	return associatedProperty(host, key: keyPath.utf8Start, initial: { host in
+		host.value(forKeyPath: String(describing: keyPath)) as? T ?? placeholder()
+		}, setter: setter)
+}
+
+/// Attaches a `MutableProperty` value to the `host` object under `key`. The property is
+/// initialized with the result of `initial`. Changes on the property's producer are
+/// monitored and written to `setter`.
+///
+/// This can be used as an alternative to `DynamicProperty` for creating strongly typed
+/// bindings on Cocoa objects.
+public func associatedProperty<Host: AnyObject, T>(_ host: Host, key: UnsafeRawPointer, initial: (Host) -> T, setter: @escaping (Host, T) -> (), setUp: (MutableProperty<T>) -> () = { _ in }) -> MutableProperty<T> {
```

Comment:
Implemented in https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3210/commits/7e369e5d19bff97080a08bdc4dbfbc1cc628b456#diff-448eaf1c1acb3e1c0522257283b94f72R22.